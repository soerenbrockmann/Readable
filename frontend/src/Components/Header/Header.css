.header {
  display: flex;
  padding: 1%;
  background-color: #e0e0e0;
  font-size: 1em;
  flex-wrap: wrap;
}

.header > div:nth-child(2) {
  margin-left: auto;
}
