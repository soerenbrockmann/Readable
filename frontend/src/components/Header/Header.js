import React from "react";
import { Link } from "react-router-dom";
import "./Header.css";
import Logo from "../Logo";
import Login from "../../containers/Login";

const Header = () => {
  return (
    <div className="header">
      <div>
        <Logo />
      </div>
      <div>
        <Link to="/post/create" className="link">
          Create Posts
        </Link>
      </div>
      <div>
        <Login />
      </div>
    </div>
  );
};

export default Header;
