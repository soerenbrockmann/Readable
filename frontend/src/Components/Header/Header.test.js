import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

describe("<Header />", () => {
  const header = shallow(<Header />);

  it("renders a logo", () => {
    expect(header.find("Logo").length).toBe(1);
  });

  it("renders properly", () => {
    expect(header).toMatchSnapshot();
  });
});
