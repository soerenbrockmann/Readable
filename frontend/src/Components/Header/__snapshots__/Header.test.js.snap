// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Logo />
</div>,
        <div>
          <Link
                    className="link"
                    replace={false}
                    to="/post/create"
          >
                    Create Posts
          </Link>
</div>,
        <div>
          <Connect(Login) />
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Logo />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="link"
            replace={false}
            to="/post/create"
>
            Create Posts
</Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Create Posts",
            "className": "link",
            "replace": false,
            "to": "/post/create",
          },
          "ref": null,
          "rendered": "Create Posts",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Login) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Logo />
</div>,
          <div>
            <Link
                        className="link"
                        replace={false}
                        to="/post/create"
            >
                        Create Posts
            </Link>
</div>,
          <div>
            <Connect(Login) />
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Logo />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="link"
              replace={false}
              to="/post/create"
>
              Create Posts
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Create Posts",
              "className": "link",
              "replace": false,
              "to": "/post/create",
            },
            "ref": null,
            "rendered": "Create Posts",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Login) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
